### DISCUSSION TITLE
Test setup doesn't seem to execute in the proper order.

### DISCUSSION THREAD

**Discussion by ShawnTheBeachy** [thomhurst/TUnit#1253] [Q&A] [answered]
According to the [docs here](https://thomhurst.github.io/TUnit/docs/tutorial-extras/setup), setup methods are executed bottom-up (base classes, then inheriting). However, that doesn't seem to be the case. It instead seems to be alphabetical(?). And within the same class, it seems to follow the method order rather than using the `Order` parameter when specified. I've attached a sample project showing the issue. If you rename `SetupA` to `SetupC` and swap the order of the methods in `SetupD`, the test passes even though it shouldn't.

[TUnitTests.zip](https://github.com/user-attachments/files/17817520/TUnitTests.zip)

**Comment by thomhurst** [comment, marked as answer]
Ah only instance methods will execute bottom up.

Static ones aren't constrained to an object/inheritance and so those must be manually ordered.

But yes, the ordering via the Order property has been knocked out so I'll sort that!

**Reply by ShawnTheBeachy** [reply]
That makes sense. Thanks!

**Reply by thomhurst** [reply]
Ordering by the order property should be resolved in 0.3.34
